src/view/ui.py

```diff
@@ -21,19 +21,16 @@ def menuFromDict(acts, activation_function=None, menu=None):
     """
     if menu is None:
         menu = QtWidgets.QMenu()
-    for a in acts:
-        if a is None:
-            menu.addSeparator()
-        elif isinstance(a, dict):
-            for suba, subacts in a.items():
-                submenu = menu.addMenu(suba)
-                menuFromDict(subacts, activation_function, submenu)
-        else:
+    for a, subacts in acts.items():
+        if len(subacts) == 0:
```

Comment:
Just little tip if you want, to check if a list is empty, you can just make:
    
    if not my_list:
        # DO STUFF

Reply:
ok nice !